### ISSUE TITLE
Bookmarking for offline data

### ISSUE THREAD

**Issue by ShirleyNekoDev** [HPI-de/hpi-flutter#91]
**Blocked by**
- [ ] #13 - Requires local database

**Description**

As a user, I'd like to bookmark a course, news or info bit to find it again later.
It would be nice to have the bookmarked page offline available.

**Requirements**
This would require:
- [ ] an bookmarking action on each of the entry pages.
- [ ] an overview page of each bookmarked page.
- [ ] actions to view and remove (delete from local store/remove bookmark) a bookmarked page.

**Implementation details**
It would be possible to expand the tagging system for bookmarking purposes. If we divide tags in local and cloud/given tags and use page type and bookmarked as tags, we would have a simple system to find stored data.
Furthermore, non-bookmarked cached entries could be pruned via time based strategies to keep the local database small.

**Comment by ShirleyNekoDev** [comment]
This will be specified via further tickets.
